pkg/webservice/handlers.go

```diff
@@ -532,6 +532,9 @@ func getApplicationHistory(w http.ResponseWriter, r *http.Request) {
 	// start and we cannot shortcut the loop using a break, we must finish iterating
 	records := imHistory.GetRecords()
 	result := getAppHistoryDAO(records)
```

Comment:
how about changing the return value of `getAppHistoryDAO` from `nil` to empty array? The advantages of using empty array are shown below:

1. avoid ugly nil check
2. easy to test (just test the return value of function `getAppHistoryDAO`)

Reply:
> This can be accomplished in a follow-up I believe.

that is ok to me

@lowc1012 please file a jira to trace that refactor :)

Reply:
@chia7712 @pbacsko
I created a new ticket.
https://issues.apache.org/jira/browse/YUNIKORN-2412